### PR TITLE
Use Quay.io Konflux dev images in upstream bundle

### PR DESCRIPTION
**Pull request by yuumasato**
## Summary

Update master branch bundle manifests to use Quay.io Konflux dev images with `:master` tags. This ensures e2e tests always use the latest Konflux builds.

**Note**: Release branches will use stable ghcr.io images with version tags.

## Problem

E2e tests used stale content images from GitHub Actions instead of latest Konflux builds, preventing immediate testing of content changes.

## Changes

**Bundle and workflows**:
- Bundle CSV now references `quay.io/redhat-user-workloads/ocp-isc-tenant/*-dev:master`
- Removed operator/openscap/must-gather builds from GitHub Actions (now built by Konflux)
- Bundle and catalog still built by GitHub Actions

**Runtime defaults and tests**:
- Base CSV template, runtime fallbacks, and e2e test defaults updated to quay.io

**Kept as ghcr.io**:
- config/manager/* (local dev placeholders)
- config/samples/* (user-facing examples should be stable)

## Flow

1. Konflux builds images → tags with `:master`
2. GitHub Actions builds bundle (contains quay.io refs) → `ghcr.io/complianceascode/compliance-operator-bundle:latest`
3. GitHub Actions builds catalog → `ghcr.io/complianceascode/compliance-operator-catalog:latest`
4. E2e tests install from catalog → get latest Konflux `:master` images

## Test plan

After merge, verify bundle contains quay.io references:
```bash
oc image extract ghcr.io/complianceascode/compliance-operator-bundle:latest --confirm
grep "quay.io/redhat-user-workloads" manifests/*.clusterserviceversion.yaml
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by yuumasato** [comment]
## Context: Why this change is needed

### Problem with ocp4e2e and content repos

The e2e test infrastructure critically depends on the catalog defined in the content repo. When [ocp4e2e installs the operator](https://github.com/ComplianceAsCode/ocp4e2e/blob/def3a79fa5538ae415a9e9bcc22f1ef6c54f63b5/helpers/setup.go#L51), it [uses the CatalogSource from content upstream](https://github.com/ComplianceAsCode/ocp4e2e/blob/def3a79fa5538ae415a9e9bcc22f1ef6c54f63b5/helpers/utilities.go#L437), which [points to](https://github.com/ComplianceAsCode/content/blob/1102fd415381da1affd3b8a7c7d4e7a26121e4eb/ocp-resources/compliance-operator-catalog-source.yaml#L10):

```yaml
image: ghcr.io/complianceascode/compliance-operator-catalog:latest
```

**The critical issue**: This catalog was built from a bundle that referenced **outdated GitHub Container Registry images**, particularly the content image `ghcr.io/complianceascode/k8scontent:latest`. 

When developers make changes to compliance rules/profiles in the content repo and want to test them via ocp4e2e, the catalog was still deploying the **stale GitHub Actions-built content image** instead of the **latest Konflux build from master**. This meant:

1. **E2e tests weren't testing latest content**: Changes to compliance rules/profiles weren't reflected in test runs
2. **Development friction**: No automatic way to verify content changes without manual image building
3. **Duplicate/divergent builds**: Konflux was building the authoritative images, but e2e used different ones from GitHub Actions
4. **Version lock-in**: The catalog couldn't be used across different OCP cluster versions

### The solution

This PR updates the bundle manifests to reference **Quay.io Konflux dev images with `:master` tags**:
- `quay.io/redhat-user-workloads/ocp-isc-tenant/compliance-operator-content-dev:master` (the key fix)
- `quay.io/redhat-user-workloads/ocp-isc-tenant/compliance-operator-dev:master`
- `quay.io/redhat-user-workloads/ocp-isc-tenant/compliance-operator-openscap-dev:master`

These images are automatically updated by Konflux on every merge to master in both the compliance-operator and content repos.

**Result**: E2e tests in ocp4e2e now automatically use the latest content changes from master, making content development testable immediately and the catalog version-agnostic across OCP versions.

**Comment by abushkin-redhat** [comment]
Hey @yuumasato, looks great! A couple of findings from Claude's `/code-review`: 

1. `bundle-hack/update_csv.go:209,215` — In `replaceImages()`, the `contentImage` fields in alm-examples are set to `newPullSpecs["RELATED_IMAGE_OPERATOR"]` instead of `newPullSpecs["RELATED_IMAGE_PROFILE"]`. This means downstream bundle builds would have the CSV examples pointing to the operator image rather than the content image, which would cause profile parsing failures if anyone copies those examples.
2. `config/manager/deployment.yaml:55-59` and `pkg/utils/images.go:17-19` — The PR removes the GHCR push workflows for operator, openscap, and must-gather, but `config/manager/deployment.yaml` (used by `make deploy-local`) and the hardcoded fallback defaults in `pkg/utils/images.go` still reference `ghcr.io/complianceascode/*:latest`. Once the workflows are gone, these images will stop receiving updates, so local dev deployments and any edge case where `RELATED_IMAGE_*` env vars are unset will silently use stale images. The base CSV template at `config/manifests/bases/compliance-operator.clusterserviceversion.yaml` has the same issue.
3. `tests/e2e/cel_tests/main_test.go:1018,1052` — In `TestCustomRuleCascadingStatusUpdate`, the CEL expression checks `pod.spec.securityContext.runAsNonRoot == true` but the `FailureReason` says "Pod(s) found without resource limits" and the TailoredProfile rationale says "Pods should have resource limits for stability" — looks like a copy-paste from a different rule.
4. `Makefile:643` — The `e2e-cel` target's help comment says "Run profile parsing end-to-end tests" — copy-pasted from the `e2e-parsing` target above it. Should say something like "Run CEL CustomRule end-to-end tests".

**Comment by yuumasato** [comment]
Thanks for review:

**#1 (bundle-hack/update_csv.go)**: This is a pre-existing bug in the Konflux bundle build script. Our PR doesn't use that script - we build the upstream bundle via `bundle.Dockerfile` which just packages the manifests as-is. This bug should be fixed separately for downstream/Konflux builds.

**#2 (config/manager/deployment.yaml and image references)**: 

The workflow removals are intentional - Konflux now builds those images. However, the impact differs by file:

- **config/manager/deployment.yaml**: Not a problem. The ghcr.io references are placeholders replaced via sed during `make deploy-local` (which builds images locally via `image-to-cluster` dependency). Local dev doesn't rely on ghcr.io having fresh `:latest` builds.

- **pkg/utils/images.go**: Updated to quay.io :master for **master branch**. These are runtime fallbacks when RELATED_IMAGE_* env vars are unset. On master, using quay.io :master ensures dev/test environments get latest builds. For release branches, we'll manually update these to ghcr.io :vX.Y.Z for stability.

- **config/manifests/bases/compliance-operator.clusterserviceversion.yaml**: Updated to quay.io :master for **master branch upstream bundles**. This ensures e2e tests (the original problem this PR solves) always use the latest Konflux-built content images. Release branches will use ghcr.io versioned images.

The approach: **master = bleeding edge (quay.io :master), releases = stable (ghcr.io :vX.Y.Z)**

**#3 and #4**: These are issues not related to this PR, they should be addressed separately.

**Review by vickeybrown — DISMISSED** [review]

**Comment by yuumasato** [comment]
Rebased to latest master to see if we can clear out the failure on
```
 === NAME  TestNonExistentDeprecatedProfile
    main_test.go:879: expected error message to be from failed profile deprecation check
2026/07/13 12:29:31 Waiting for run of test-missing-pod-scan compliancescan (RUNNING)
--- FAIL: TestNonExistentDeprecatedProfile (25.02s)
```

File: images/testcontent/Dockerfile.ci [hunk 1]
```diff
@@ -1 +1 @@
-FROM ghcr.io/complianceascode/k8scontent:latest
+FROM quay.io/redhat-user-workloads/ocp-isc-tenant/compliance-operator-content-dev:master
```
Comment (yuumasato) [inline images/testcontent/Dockerfile.ci:1]: Added this change so that `TestNonExistentDeprecatedProfile` passes.

**Comment by Vincent056** [comment]
Closed content#14968 in favor of this. Verified: `content-dev:master` is a true floating tag, the image layout matches k8scontent (datastream + CEL content at `/`), so it covers CEL e2e too.

/lgtm

**Review by Vincent056 — APPROVED** [review]
/lgtm
